### DISCUSSION TITLE
borg mount not available: loading FUSE support failed

### DISCUSSION THREAD

**Discussion by pin00b** [borgbase/vorta#836] [General]
**Describe the bug**
Trying to mount an archive from a given repo fails, ERROR: 
borg mount not available: loading FUSE support failed (ImportError: dlopen(/Applications/Vorta.app/Resources/borg-
dir/llfuse.cpython-37m-darwin.so,2): Library not loaded: @loader _path/libosxfuse.2.dylib
Referenced from:
/Applications/Vorta.app/Contents/Resources/borg-dir/llfuse.cpython-37m-darwin.so
Reason:
no suitable image found
Did find:
file system relative paths not allowed in hardened programs

**Desktop:**
- OS: macOS 11.0.1
- Vorta version: 0.7.1, Borg: 1.1.13
- Installed from: DMG/binary
- FUSE: 4.0.4

If appropriate include logs. Can be found in Main Window > Misc Tab > Log.
--> 
2020-12-04 17:21:18,174 - vorta.borg.borg_thread - ERROR - borg mount not available: loading FUSE support failed [ImportError: dlopen(/Applications/Vorta.app/Contents/Resources/borg-dir/llfuse.cpython-37m-darwin.so, 2): Library not loaded: @loader_path/libosxfuse.2.dylib
  Referenced from: /Applications/Vorta.app/Contents/Resources/borg-dir/llfuse.cpython-37m-darwin.so
  Reason: no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs]

Any help greatly appreciated! Thx!

**Comment by samuel-w** [comment]
I believe this is a duplicate of https://github.com/borgbase/vorta/issues/733.

**Comment by pin00b** [comment]
> I believe this is a duplicate of #733.

Similar, but not identical, since in my case it states the library not to be loaded, in #733 the error is the following:
[ImportError: No module named 'llfuse']

**Comment by m3nu** [comment]
FUSE support was recently removed from Homebrew/macOS. We now maintain a custom Tap for it:

https://github.com/borgbackup/homebrew-tap

So most likely we won't include Borg in future Vorta releases, since it got too complex to include FUSE. Too sensitive to different FUSE versions.

**Reply by rupertpaulson** [reply]
I agree with pin00b's comment. Do you really mean "we won't include Borg in future Vorta releases" or rather "we won't include FUSE in future Vorta releases"? I'm a bit confused...

**Reply by m3nu** [reply]
It's getting harder to ship a fat Borg binary together with Vorta due to FUSE support. So installing it from Borg's own Homebrew Tap is currently the most reliable option. All documented here: https://vorta.borgbase.com/install/macos/

We will probably keep bundling Borg with Vorta anyways, since mounting is only a minor feature. Everything else will still work and help non-Homebrew users to get started quickly.

**Comment by pin00b** [comment]
> FUSE support was recently removed from Homebrew/macOS. We now maintain a custom Tap for it:
> 
> https://github.com/borgbackup/homebrew-tap
> 
> So most likely we won't include Borg in future Vorta releases, since it got too complex to include FUSE. Too sensitive to different FUSE versions.

Maybe I am too stupid to understand it properly: I installed both Vorta and FUSE via DMG and .pkg respectively. Should I better install both via Homebrew and use FUSE from the tap you support? - Did you plan to remove FUSE support from Vorta in the future? Because Vorta without Borg doesn't make any sense at all!? No? - What would a possibility to mount archives or even repos look like in the future?

**Comment by m3nu** [comment]
If you want FUSE support, you need to install Borg via PyPi or our Tap above. The Tap is rather new and not documented yet.

The official Homebrew Borg doesn't support FUSE any more and the built-in version only works with specific FUSE versions.

We don't plan on removing FUSE support, but we do have to follow developments upstream.

**Comment by dnszero** [comment]
In the Vorta GUI (after installing via DMG), the Archives tab still has a link to go download and install FUSE.  ...which is what I did before getting the same error as pin00b and landing here.

Just a reminder to change that message/link for future updates.

**Comment by m3nu** [comment]
There are many changes to macFUSE at the moment. We need to wait how it plays our, also in relation to macOS arm64.

**Comment by m3nu** [comment]
Generally best to follow the install instructions on the website and use Homebrew. We recently added a Borg tap with working FUSE support: https://vorta.borgbase.com/install/macos/

**Comment by arnonel** [comment]
Followed instructions above and still get error: "No suitable image found" and "file system relative paths not allowed in hardened programs"

**Comment by TensorBlast** [comment]
Quite pathetic that this issue still exists in Vorta 0.7.5 and Borg at this date. Enough time has passed to have had this issue resolved. This is why free software is often just junkware.

**Reply by m3nu** [reply]
Open source also means that you can help improve it. So I’m looking forward to your contribution to improve this issue or tell us what we can change. 😀

**Reply by TensorBlast** [reply]
Maintainer couldn't help and basically said to sort it out myself. Promptly uninstalled borg, vorta and happy to have my mental peace back.

**Comment by TensorBlast** [comment]
Had I the skill to reverse engineer the issue and “improve” it, I would have done so already. I’ve reached out to the community and created an “Issue” providing all details that I could in the hopes that I am helping in some way.
Your comment comes off as standoffish and implies that Im not trying to help.

Back to the issue - is it not reproducible on your end (assuming you’re using a M1 Mac)?

**Reply by m3nu** [reply]
Myself, I don't have M1 hardware for testing, but others have tested Borg and Vorta successfully on M1 shortly after it was released. E.g. here: https://github.com/borgbackup/homebrew-tap/issues/6

**Reply by TensorBlast** [reply]
Hardly call that a success. Your own link shows a litany of issues still unresolved despite the latest fix with people taping over holes to desperately to call it "fixed"

**Reply by m3nu** [reply]
No need to reverse engineer anything. Our source code is open. 😁

Also, you looked into the issue in detail and found the missing path. So you did make an important contribution in #1001 and #1002. Glad we got this solved.
